src/merkle/mmr/partial.rs

```diff
@@ -0,0 +1,81 @@
+//! A partial view of the MMR, useful to maintain an update proof for a single value.
+use super::{super::Vec, full::nodes_in_forest, Rpo256, Word};
+
+/// A partial view of the MMR.
+///
+/// Data structure used to maintain the proof of a single value up-to-date. Only the data relevant
+/// to `value` is stored, meaning the path from the leaf to the root, and the peaks in the MMR that
+/// may be merged with the current peak. Required updates to the structure are very infrequent, and
+/// require one value, the new entry to the right of the current peak added duraing a merge.
+#[derive(Debug, Clone, PartialEq)]
+pub struct MmrPartial {
+    /// The MMR peaks of interest for this partial view, this includes all the peaks to the left of
+    /// the current root, and the root itself.
+    pub(crate) peaks_partial: Vec<Word>,
+
+    /// The representation of all the peaks in this partial view.
+    pub(crate) forest_partial: usize,
+
+    /// The Merkle opening, starting from the value's sibling up to and excluding the root of the
+    /// responsible tree.
+    pub(crate) path: Vec<Word>,
+
+    /// The value this opening corresponds to
+    pub(crate) value: Word,
+}
+
+impl MmrPartial {
+    // CONSTRUCTORS
+    // ============================================================================================
+
+    pub fn new(
+        peaks_partial: Vec<Word>,
+        forest_partial: usize,
+        path: Vec<Word>,
+        value: Word,
+    ) -> Option<MmrPartial> {
+        if peaks_partial.is_empty() || peaks_partial.len() as u32 != forest_partial.count_ones() {
+            None
+        } else {
+            Some(MmrPartial {
+                peaks_partial,
+                forest_partial,
+                path,
+                value,
+            })
+        }
+    }
```

Comment:
This looks more like a `Result` than `Option` as what we do is validate the arguments. Maybe add a comment on why we don't want an empty peaks set? Also, should this set be unique? If yes, we should probably check that as well, or use a `BTreeSet` that will have that guaranteed by definition

Reply:
1. changed to result
2. added comment
3. there is no guarantee on the values being unique